concurrency/limited_concurrency_singleflight_test.go

```diff
@@ -0,0 +1,133 @@
+package concurrency
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+func TestLimitedConcurrencySingleFlight_ForEachNotInFlight_ConcurrencyLimit(t *testing.T) {
+	t.Parallel()
+
+	var (
+		ctx            = context.Background()
+		pool           = NewWorkerPool(10)
+		workersWait    chan struct{}
+		workersToStart sync.WaitGroup
+	)
+
+	forEachNotInFlight := func(tokens []int, f func(context.Context, string) error) {
+		stringTokens := make([]string, len(tokens))
```

Comment:
nit: Seems a bit unnecessary to pass ints in just to convert them to strings.

Reply:
my thought was that it's easier to read if they are int slices not string slices. I will change to variadic strings. that should work for both